### ISSUE TITLE
[develop] New grain loading stacktrace introduced due to new zfs grains

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt#44447]
### Description of Issue/Question
The following stacktrace occurs and is logged many times when running at the head of develop. Here's a quick example using salt-call:
```
# salt-call --local test.ping
[CRITICAL] Failed to load grains defined in grain file zfs.zfs in function <function zfs at 0x7f438e6b1c08>, error:
Traceback (most recent call last):
  File "/root/SaltStack/salt/salt/loader.py", line 752, in grains
    ret = funcs[key]()
  File "/root/SaltStack/salt/salt/grains/zfs.py", line 109, in zfs
    grains = salt.utils.dictupdate.update(grains, _zfs_support(), merge_lists=True)
  File "/root/SaltStack/salt/salt/grains/zfs.py", line 75, in _zfs_support
    if _zfs_fuse('available') and (_zfs_fuse('status') or _zfs_fuse('start')):
  File "/root/SaltStack/salt/salt/grains/zfs.py", line 74, in <lambda>
    _zfs_fuse = lambda f: __salt__['service.' + f]('zfs-fuse')
KeyError: 'service.available'
```

### Setup
Run on the latest develop, this example is on Ubuntu 14, but I have seen it in test logs other places.

### Versions Report
This is running at the head of the develop branch:
```
# salt --versions
Salt Version:
           Salt: 2017.7.0-796-g5aaea7f

Dependency Versions:
           cffi: 1.11.2
       cherrypy: unknown
       dateutil: 2.6.1
      docker-py: 2.5.1
          gitdb: 2.0.2
      gitpython: 2.1.5
          ioflo: 1.5.5
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.3
      pycparser: 2.18
       pycrypto: Not Installed
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: 0.3.8
         PyYAML: 3.12
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: 2.0.3
        timelib: 0.2.4
        Tornado: 4.5.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
         locale: UTF-8
        machine: x86_64
        release: 4.9.15-x86_64-linode81
         system: Linux
        version: Ubuntu 14.04 trusty
```

@sjorge This is due to some of the changes you've been making recently. Would you mind taking a look?

**Comment by sjorge** [comment]
I’m about to head to bed, I will take a look at this tomorrow.

The issues seems to be that the service module is not available at this stage. Should not be too hard too fix.

**Comment by sjorge** [comment]
Quick question, googling around it seems ZFS-fuse has mostly been abandoned in favor of zfsonlinux.

Would it be OK to drop support for that? This particular check has cause a bit of trouble in the past, it was also tripping up OpenBSD.

If we prefer to keep it a simple check if the zfs-fuse binary exist should be a acceptable replacement check as we only check if the service.

**Comment by sjorge** [comment]
@rallytime is the problem fixed with:
```patch
--- a/salt/grains/zfs.py
+++ b/salt/grains/zfs.py
@@ -70,10 +70,8 @@ def _zfs_support():
             on_supported_platform = _check_retcode('ls /sys/module/zfs')

         # NOTE: fallback to zfs-fuse if needed
-        if not on_supported_platform:
-            _zfs_fuse = lambda f: __salt__['service.' + f]('zfs-fuse')
-            if _zfs_fuse('available') and (_zfs_fuse('status') or _zfs_fuse('start')):
-                on_supported_platform = True
+        if not on_supported_platform and salt.utils.path.which('zfs-fuse'):
+            on_supported_platform = True

     # Additional check for the zpool command
     if on_supported_platform and salt.utils.path.which('zpool'):
```

**Comment by rallytime** [comment]
@sjorge Yes, that fixes it. Thank you!

And no worries - this can wait until tomorrow.

> Would it be OK to drop support for that? This particular check has cause a bit of trouble in the past, it was also tripping up OpenBSD.

I think generally we try to phase support out for things like that gradually by putting them on a deprecation path. Would that be possible instead of fully dropping support for something people might be using?

**Comment by sjorge** [comment]
I created the PR as I already pushed the little diff to my own repo.

Not sure how to gracefully phase out zfs-fuse support, I kept the original support but I could not even get zfs-fuse working at the time I overhauled the module.

I only some basic tests on FreeBSD, linux using zfsonlinux and illumos. Although there should be nothing from stopping this from working against zfs-fuse, aside from maybe one or two commands not existing in the very old zfs implementations.
